temporal-test-server/src/main/java/io/temporal/internal/testservice/StateMachines.java

```diff
@@ -686,6 +676,7 @@ private static void scheduleNexusOperation(
             .setInput(attr.getInput())
             .setScheduleToCloseTimeout(attr.getScheduleToCloseTimeout())
             .putAllNexusHeader(attr.getNexusHeaderMap())
+            .putNexusHeader("Request-Timeout", String.valueOf(data.requestTimeout.toMillis()))
```

Comment:
Yeah, `data.requestTimeout` is just the duration, it's not a timer. I put it there because it was convenient for the constant, but I could also put it somewhere else.

Reply:
The important part is that the request timeout gives the actual request timeout, it should be in sync with the time the server is willing to wait until it times out the request.
There's two options: 
- If you start a timer to track this timeout (via `timeoutNexusRequest`) when the request is scheduled and a worker takes 4 seconds to pick up the request, the header should have a value of 6.
- If you `timeoutNexusRequest` when the worker picks it up, using 10 seconds for the header value is okay.

AFAICT, this is not the case. And I don't see where you reset the nexus request timeout on retry but I didn't read the code very thoroughly.